ui_framework/components/tool_bar/tool_bar_search_box.test.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { render, mount } from 'enzyme';
+import sinon from 'sinon';
+import { commonHtmlProps } from '../../test/common_html_props';
+
+import {
+  KuiToolBarSearchBox,
+} from './tool_bar_search_box';
+
+describe('KuiToolBarSearchBox', () => {
```

Comment:
Hmm, well like I said earlier I don't feel strongly either way, but playing devils advocate here, describe blocks can make readability worse.

Take this:

```
describe('warning alert', () => {
  describe('handles warning', () => {
    test('with no children', () => ...);
    test('with children', () => ..._);
  });
});
```

Rather than reading a sentence to determine what a test is testing, you have to jump around to put the sentence together, and remember the pieces of the sentence.  So if you are scrolling through a big test file, all you see is that this tests `with children` and you have to scroll up to prepend each piece of the sentence together to finally figure out what it's testing.
`with children`.... `handles warning`.... `warning alert`... => `warning alert handles warning with children`.

Now arguing the other side, I like less repetition, easier to refactor, but still less readable, IMO.

There is a balance for each situation, which a rule can't determine for us.